plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -51,20 +50,55 @@ function arrayStringifyHelper(arr: Array<unknown>): string {
   return '[' + arr.toString() + ']';
 }
 
-// Helper function to get a low cardinality command name from the full text query
-function getCommandFromText(text?: string): string {
-  if (!text) return 'unknown';
-  const words = text.split(' ');
-  return words[0].length > 0 ? words[0] : 'unknown';
+/**
+ * Helper function to get a low cardinality span name from whatever
+ * info we have about the query.
+ *
+ * This is tricky, because we don't have most of the information (table name,
+ * operation name, etc) the spec recommends using to build a low-cardinality
+ * value w/o parsing. So, we use db.name and assume that, if the query's a
+ * named prepared statement, those `name` values will be low cardinality.
+ * If we don't have a named prepared statement, we try to parse an operation
+ * (despite the spec's warnings).
+ *
+ * @params query The information we have about the query, typed to reflect
+ *   only the validation we've actually done on the `client.query()` args.
+ */
+export function getQuerySpanName(
+  databaseName: string | undefined,
+  queryConfig?: { text: string; name?: unknown }
+) {
+  if (!queryConfig) return PgInstrumentation.BASE_SPAN_NAME;
+
+  // Either the name of a prepared statement; or an attempted parse
+  // of the SQL command, normalized to uppercase; or unknown.
+  const command =
+    typeof queryConfig.name === 'string' && queryConfig.name
+      ? queryConfig.name
+      : queryConfig.text.split(' ')[0].toUpperCase() || 'unknown';
+
+  return `${PgInstrumentation.BASE_SPAN_NAME}:${command} ${databaseName ?? ''}`;
```

Comment:
Now that `command` can contain spaces and the database name is also appended, it might create span names such as `pg.query:find users users` which I guess are not straight forward to read.

I read your awesome description above about the complexity of building the span name in this case. I am good with keeping it as it improves the current state, but wonder if you considered any alternatives which might be more readable to users or better align with the spec suggestion `<db.operation> <db.name>`

Reply:
Hmm, it's a good question about alternatives. I'm not sure I have any great ideas, except maybe to put the name in quotation marks? So we'd have span names like:

```
pq.query:SELECT api-server-db
pg.query:"find user by id" api-server-db
pg.query:"findUser" api-server-db
```

I'm not an Otel expert, though, so I'm not sure if quotation marks in the span name are legal or, even if legal, are likely to cause problems for implementations.

Reply:
Yeah I am not sure it's a good idea. the spec doesn't say a lot about it.
Let's keep it as is for now and change in the future if we get feedback